src/algorithms/all_neighbors.c

```
@@ -121,14 +120,14 @@ EntityID AllNeighborsCtx_NextNeighbor
 			// backtrack
 			ctx->current_level--;
 			dest_id = array_pop(ctx->visited);
-			int res = HashTableDelete(ctx->visited_nodes, (void*)(dest_id));
-			ASSERT(res == DICT_OK);
+			const void *res = hashmap_delete_with_hash(ctx->visited_nodes, NULL, dest_id);
+			ASSERT(res != NULL);
```

Cppcheck (2.10-2)

[error] Null pointer dereference

(nullPointer)